src/Private/Version.ps1

```diff
@@ -0,0 +1,127 @@
+PS> Get-PodeVersion
+
+Returns '[develop branch]'.
+
+.NOTES
+This function assumes that $moduleManifest is a hashtable representing the loaded module manifest, with a key of ModuleVersion.
+
+#>
+function Get-PodeVersion {
+    $moduleManifest = Get-ModuleManifest
+    if ($moduleManifest.ModuleVersion -ne '$version$') {
+        return "v.$($moduleManifest.ModuleVersion)"
+    }
+    else {
+        return '[develop branch]'
+    }
+}
+
+<#
+.SYNOPSIS
+Tests if the Pode module is from the development branch.
+
+.DESCRIPTION
+The Test-PodeDevelop function checks if the Pode module's version matches the placeholder value ('$version$'), which is used to indicate the development branch of the module. It returns $true if the version matches, indicating the module is from the development branch, and $false otherwise.
+
+.PARAMETER None
+This function does not accept any parameters.
+
+.OUTPUTS
+System.Boolean
+Returns $true if the Pode module version is '$version$', indicating the development branch. Returns $false for any other version.
+
+.EXAMPLE
+PS> $moduleManifest = @{ ModuleVersion = '$version$' }
+PS> Test-PodeDevelop
+
+Returns $true, indicating the development branch.
+
+.EXAMPLE
+PS> $moduleManifest = @{ ModuleVersion = '1.2.3' }
+PS> Test-PodeDevelop
+
+Returns $false, indicating a specific release version.
+
+.NOTES
+This function assumes that $moduleManifest is a hashtable representing the loaded module manifest, with a key of ModuleVersion.
+
+#>
+function Test-PodeDevelop {
+    return (Get-ModuleManifest).ModuleVersion -eq '$version$'
+}
+
+
+<#
+.SYNOPSIS
+Tests if the current PowerShell version is considered End-of-Life (EOL).
+
+.DESCRIPTION
+The Test-PSVersionEOL function checks if the PowerShell session is running on the Core edition and if the version is earlier than $moduleManifest.PrivateData.PwshCoreVersionUntested. If both conditions are met, it indicates that the PowerShell version is EOL. A warning message is displayed, noting that while Pode should still function, it has not been tested on EOL versions.
+
+.PARAMETER None
+This function does not accept any parameters.
+
+.EXAMPLE
+PS> Test-PSVersionEOL
+
+If running on PowerShell Core version earlier than $moduleManifest.PrivateData.PwshCoreVersionUntested, you will see a warning message indicating the version is EOL.
+
+#>
+function Test-PSVersionEOL {
+    $moduleManifest = Get-ModuleManifest
+    if ($moduleManifest.ModuleVersion -ne '$version$') {
+        # Check if the PowerShell edition is Core and the version is earlier than the last version tested
+        if ($PSVersionTable.PSEdition -eq 'Core' -and $PSVersionTable.PSVersion -lt [version]$moduleManifest.PrivateData.PwshCoreVersionUntested) {
+            Write-Host "This PowerShell version $($PSVersionTable.PSVersion) was EOL when Pode $(Get-PodeVersion) was released. Pode should work but has not been tested." -ForegroundColor Yellow
+            return $true
+        }
```

Comment:
This check will start reporting PS7.2 errornously if we do a release of Pode between May-November 2024, as PS7.3 will be set in the PwshCoreVersionUntested, and 7,2 is less than 7.3 😄 

I'd also use `Write-PodeHost`, and put a `[WARNING]` at the start of the message.

Reply:
You are pointing to an issue I was not really thinking through. Maybe it's easy to compare the running version with any unsupported one

Reply:
One way could be to have the property in the psd1 be a comma separated list of EOL versions, ie:
```powershell
$eol = Invoke-RestMethod -Uri 'https://endoflife.date/api/powershell.json' -Headers @{Accept = 'application/json' }
$cycles = ($eol | Where-Object { [datetime]$_.eol -lt [datetime]::Now }).cycle -join ','
```
And then the property in the psd1 be:
```powershell
PwshVersionsUntested = '$versionsUntested$'

# remapped:
PwshVersionsUntested = '7.1,7.0,6.2,6.1,6.0,5.0,4.0,3.0,2.0,1.0'
```

And then in `Test-PodeVersionPwshEOL` compare the `Major.Minor` of the current console against the above list:
```powershell
function Test-PodeVersionPwshEOL {
    $moduleManifest = Get-PodeModuleManifest
    if ($moduleManifest.ModuleVersion -eq '$version$') {
        return $false
    }

    $psVersion = $PSVersionTable.PSVersion
    $eolVersions = $moduleManifest.PrivateData.PwshVersionsUntested -split ','
    $isEol = "$($psVersion.Major).$($psVersion.Minor)" -in $eolVersions

    if ($isEol) {
        Write-PodeHost "[WARNING] The running version of PowerShell, $($PSVersionTable.PSVersion), was EOL when Pode $(Get-PodeVersion) was released. Pode should work but has not been tested on this version of PowerShell." -ForegroundColor Yellow
    }

    return $isEol
}
```

This should automagically cover PS7.3, then 7.2, and even 5.1 in 2027.